### PR TITLE
Full auto engine

### PR DESCRIPTION
**Pull request by adsau59**
- rework to use fishyqr instead of ocr
- added blob detection to find fishyqr addon automatically
- install required addon automatically
- optimized controls for better user experience 

only change in semifisher : https://github.com/fishyboteso/fishyboteso/commit/c05355fb7737d255c5b3ff4e43687ac0d44b40e0

### CODE REVIEW

File: fishy/constants.py [hunk 1]
```diff
@@ -1,3 +1,6 @@
 apiversion = 2
 chalutier = ("Chalutier", "https://github.com/fishyboteso/Chalutier/raw/619b4ab0b8ff91746afda855542e886d27b7a794/Chalutier_1.1.2.zip", 112)
 lam2 = ("LibAddonMenu-2.0", "https://www.esoui.com/downloads/dl7/LibAddonMenu-2.0r32.zip", 32)
+
+fishyqr = ("FishyQR", "https://github.com/fishyboteso/FishyQR/files/6329586/FishyQR.zip", 1)
```
Comment (SemjonKerner) [inline fishy/constants.py:5]: for sake of sanity, can you please use the correct versions? 100 for fishyqr and 30 for libgps

File: fishy/constants.py [hunk 2]
```diff
@@ -1,3 +1,6 @@
...
+libgps = ("LibAddonMenu-2.0", "https://cdn.esoui.com/downloads/file601/LibGPS_3_0_3.zip", 1)
```
Comment (SemjonKerner) [inline fishy/constants.py:6]: the name is a leftover of copy pasting

**Review by SemjonKerner — CHANGES_REQUESTED** [review]
You are changing the requirements.txt anyway, could you maybe use d3dshot to get the screenshot instead? It's much faster.
See https://github.com/SemjonKerner/fishyboteso/blob/fishylight/qrtest/fishylight.py#L10
Lines 10, 13 and 30

I did not read the fullfisher files. Can't spend the time understanding them.

**Comment by adsau59** [comment]
> You are changing the requirements.txt anyway

- Added modules for qr code reading, and a library to get mouse events, 

> could you maybe use d3dshot to get the screenshot instead? It's much faster.

Yeah true, but image grabbing is abstracted in WindowServer which isn't related to full auto... We can create a new PR to implement that optimization

**Pull request merged**